sw/device/silicon_creator/lib/drivers/watchdog_unittest.cc

```diff
@@ -5,72 +5,89 @@
 #include "sw/device/silicon_creator/lib/drivers/watchdog.h"
 
 #include "gtest/gtest.h"
-#include "sw/device/lib/base/mmio.h"
 #include "sw/device/silicon_creator/lib/base/mock_abs_mmio.h"
+#include "sw/device/silicon_creator/lib/base/mock_sec_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/mock_otp.h"
 
 #include "aon_timer_regs.h"
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
 #include "pwrmgr_regs.h"
 
 namespace watchdog_unittest {
 namespace {
-
-constexpr uint32_t kAonTimerRate = 200000;
+using ::testing::Return;
 
 class WatchdogTest : public mask_rom_test::MaskRomTest {
  protected:
   uint32_t pwrmgr_ = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR;
   uint32_t wdog_ = TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR;
-  mask_rom_test::MockAbsMmio mmio_;
+  mask_rom_test::MockAbsMmio abs_;
+  mask_rom_test::MockSecMmio sec_;
+  mask_rom_test::MockOtp otp_;
 };
 
-TEST_F(WatchdogTest, InitializeEnable) {
-  constexpr uint32_t kTimeoutMs = 100;
-  constexpr uint32_t kBiteThreshold = kTimeoutMs * (kAonTimerRate / 1000);
-
-  EXPECT_ABS_WRITE32(pwrmgr_ + PWRMGR_RESET_EN_REG_OFFSET,
-                     {
-                         {PWRMGR_RESET_EN_EN_1_BIT, true},
-                     });
-  EXPECT_ABS_WRITE32(pwrmgr_ + PWRMGR_CFG_CDC_SYNC_REG_OFFSET, 1);
-
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WDOG_CTRL_REG_OFFSET, 0);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WKUP_CTRL_REG_OFFSET, 0);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WKUP_COUNT_REG_OFFSET, 0);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WDOG_COUNT_REG_OFFSET, 0);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WKUP_THOLD_REG_OFFSET, 0xFFFFFFFF);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WDOG_BARK_THOLD_REG_OFFSET, 0xFFFFFFFF);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WDOG_BITE_THOLD_REG_OFFSET,
-                     kBiteThreshold);
-  EXPECT_ABS_WRITE32(wdog_ + AON_TIMER_WDOG_CTRL_REG_OFFSET, 1);
-
-  watchdog_init(kTimeoutMs);
+TEST_F(WatchdogTest, InitializeNoOtp) {
+  constexpr std::array<lifecycle_state_t, 2> kLifecycleStates = {kLcStateTest,
+                                                                 kLcStateRma};
+  for (const auto lc : kLifecycleStates) {
+    const uint32_t kBiteThreshold = 0x12345678;
+    EXPECT_CALL(
```

Comment:
Feel free to ignore.
But I would move the duplicated code in these two tests to a method in the `WatchdogTest` class.

Reply:
Ignoring for now due to time constraints.